internal/server/runner.go

```diff
@@ -120,6 +120,7 @@ func NewRunner(awaitExplicitShutdown bool, uploadUrl string) *Runner {
 
 func NewProcedureRunner(awaitExplicitShutdown bool, uploadUrl string, srcDir string) *Runner {
 	r := NewRunner(awaitExplicitShutdown, uploadUrl)
+	os.Mkdir(srcDir, 0o755)
```

Comment:
I'd move this to `server.go` where we hash URL & extract tarball. Runner should always assume dir is prepared.